pyamgx/Matrix.pyx

```diff
@@ -182,12 +182,23 @@ cdef class Matrix:
             Array of matrix data.
         """
         cdef int n, nnz
+        
+        if hasattr(data, "__array_interface__"):
+            desc = data.__array_interface__
+        elif hasattr(data, "__cuda_array_interface__"):
+            desc = data.__cuda_array_interface__
+        else:
+            raise TypeError("Must pass array-like for data")
+            
+        cdef uintptr_t ptr = desc["data"][0]
+        
```

Comment:
Could we use the utility function [`ptr_from_array_interface`](https://github.com/shwina/pyamgx/blob/main/pyamgx/utils.pyx#L5) here instead?

Reply:
Ok, should I check for dtype here?

Reply:
Yes -- let's do that